### ISSUE TITLE
[Bug][DropDownList] Screen Readers does not read the items when the data is filterable

### ISSUE THREAD

**Issue by Yanko-Georgiev** [telerik/kendo-react#3075]
### Description

When filtering is enabled, the screen readers do not read the items correctly.

### Plan

KendoReact

### Steps To Reproduce

1. Open the following example: https://stackblitz.com/edit/react-pfjvwsbz?file=app%2Fapp.tsx
2. Turn on your screen reader(Voice Over on Mac)
3. Navigate through different items
4. Notice that are not read from the screen reader

### Screenshots or video

https://github.com/user-attachments/assets/523c95e4-b7e3-4496-a5f1-aae2ee21e9f3

### Actual Behavior

The screen reader does not read the items in the DropDownList.

### Expected Behavior

The screen reader should read the items in the DropDownList.

### Browser

Chrome

### Browser version

latest

### OS type

MacOS

Reported in TicketID: 1685550
